client/state/form/reducer.js

```diff
@@ -34,9 +27,13 @@ export default function form( state = {}, action ) {
 		} );
 	}
 
-	// Allow client-side form validation to take over error state when inputs change
-	if ( settingsActions.hasOwnProperty( action.type ) && newState.hasOwnProperty( 'errors' ) ) {
-		delete newState.errors;
+	if ( settingsActions.hasOwnProperty( action.type ) ) {
+		newState.pristine = false;
```

Comment:
The form becomes "dirty" when a setting changes, and only becomes `pristine` when it's finished saving the changes. Inputting some text in a text field doesn't call the `SET_FIELD` reducer, so what you proposed wouldn't work here.

Side note: the `SET_FIELD` name is the most confusing name ever. It changes a property of the form (such as `isSaving`), not form fields.